### PR TITLE
refactor(pipeline): refactor actionagentSvc,permissionSvc to provider

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
refactor actionagentSvc,permissionSvc to provider

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | refactor actionagentSvc,permissionSvc to provider             |
| 🇨🇳 中文    |  将actionagentSvc和permissionSvc迁移到provide            |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/tools/pipeline/providers/actionagent/interface.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package actionagent
+
+import "github.com/erda-project/erda/apistructs"
+
+type Interface interface {
+	Ensure(apistructs.ClusterInfoData, string, string) error
```
Comment (sfwn) [inline internal/tools/pipeline/providers/actionagent/interface.go:20]: add var name

Comment (sfwn) [reply internal/tools/pipeline/providers/actionagent/interface.go:20]: and comments

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/tools/pipeline/providers/actionagent/interface.go (hunk 1 above)
Comment (chengjoey) [reply internal/tools/pipeline/providers/actionagent/interface.go]: adjusted

**Review by sfwn — APPROVED** [review]

**Pull request merged**